engines/grim/bitmap.cpp

```diff
@@ -316,36 +307,41 @@ bool BitmapData::loadTile(Common::SeekableReadStream *o) {
 	o->seek(-8, SEEK_CUR);
 
 	_data = new Graphics::Surface[_numImages];
-	Graphics::PixelFormat pixelFormat = Graphics::PixelFormat(4, 8, 8, 8, 8, 0, 8, 16, 24);
+
+	Graphics::PixelFormat format_16bpp = Graphics::PixelFormat(2, 5, 5, 5, 1, 0, 5, 10, 15);
+#ifdef SCUMM_BIG_ENDIAN
+	Graphics::PixelFormat format_32bpp = Graphics::PixelFormat(4, 8, 8, 8, 8, 24, 16, 8, 0);
+#else
+	Graphics::PixelFormat format_32bpp = Graphics::PixelFormat(4, 8, 8, 8, 8, 0, 8, 16, 24);
+#endif
+
 	int width = 256;
 	int height = 256;
 
 	for (int i = 0; i < _numImages; ++i) {
-		_data[i].create(width, height, pixelFormat);
+		_data[i].create(width, height, (_bpp == 16) ? format_16bpp : format_32bpp);
+		uint8 *d = (uint8 *)_data[i].getPixels();
 		o->seek(8, SEEK_CUR);
-		if (_bpp == 16) {
-			uint32 *d = (uint32 *)_data[i].getPixels();
-			for (int j = 0; j < _width * _height; ++j) {
-				uint16 p = o->readUint16LE();
-				// These values are shifted left by 3 so that they saturate the color channel
-				uint8 b = (p & 0x7C00) >> 7;
-				uint8 g = (p & 0x03E0) >> 2;
-				uint8 r = (p & 0x001F) << 3;
-				uint8 a = (p & 0x8000) ? 0xFF : 0x00;
-				// Recombine the color components into a 32 bit RGB value
-				uint32 tmp = (r << 24) | (g << 16) | (b << 8) | a;
-				WRITE_BE_UINT32(&d[j], tmp);
+		switch (_bpp) {
+		case 16:
+#ifndef SCUMM_LITTLE_ENDIAN
+			for (int y = 0; y < _height; ++y) {
+				uint16 *d16 = (uint16 *)d;
+				for (int x = 0; x < _width; ++x) {
```

Comment:
This should be fixed for BE systems

```suggestion
				for (int x = 0; x < _width; ++x)
```